api/server/router/image/image_routes.go

```diff
@@ -67,7 +71,61 @@ func (ir *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrit
 		progressErr = ir.backend.PullImage(ctx, img, tag, platform, metaHeaders, authConfig, output)
 	} else { // import
 		src := r.Form.Get("fromSrc")
-		progressErr = ir.backend.ImportImage(ctx, src, repo, platform, tag, message, r.Body, output, r.Form["changes"])
+
+		var ref reference.Named
+		if repo != "" {
+			var err error
+			ref, err = reference.ParseNormalizedNamed(repo)
+			if err != nil {
+				return errdefs.InvalidParameter(err)
+			}
+			if _, isCanonical := ref.(reference.Canonical); isCanonical {
+				return errdefs.InvalidParameter(errors.New("cannot import digest reference"))
+			}
+
+			if tag != "" {
+				ref, err = reference.WithTag(ref, tag)
+				if err != nil {
+					return errdefs.InvalidParameter(err)
+				}
+			} else {
+				ref = reference.TagNameOnly(ref)
+			}
+		}
+
+		if len(comment) == 0 {
+			comment = "Imported from " + src
+		}
```

Comment:
Also not new, but my eye dropped on this, and wondering if we should look at this (in light of the containerd integration); I know quite some users use `docker import` to transfer / load images into a daemon. Currently (pre containerd) this discards the (registry-)digest, so while the image is (mostly) the same, it's not the same as doing a `docker pull` of the same image.

I wonder how far we get with making them the same (with the containerd integration), and in that case, we should avoid mutating the imported image; the `Imported from` may not be very relevant, and if setting this message by default may affect the imported image (making it not the same as `docker pull`).

I can create a ticket for that for a follow-up.

Reply:
I'm +1 on removing this - IMO it doesn't add any value, especially when in most cases src is `-`.
I'll remove it and open a follow up PR so we can discuss this and not block this PR.

Reply:
https://github.com/moby/moby/pull/44801